server/src/main/java/org/opensearch/index/mapper/MappingLookup.java

```diff
@@ -190,8 +194,49 @@ public void checkLimits(IndexSettings settings) {
         checkNestedLimit(settings.getMappingNestedFieldsLimit());
     }
 
+    /**
+     *
+     * @param limit the value of the setting index.mapping.total_fields_limit
+     * @param dynamicMappers mappers of the new fields detected by dynamic mapping
+     * @return true if adding new fields will cause to exceed the total fields limit
+     */
+    public boolean exceedTotalFieldsLimit(long limit, List<Mapper> dynamicMappers) {
+        if (nonMetadataFieldCount > limit) {
+            return true;
+        }
+
+        List<ObjectMapper> newObjectMappers = new ArrayList<>();
```

Comment:
This method and the one below look expensive to be happening on every new field addition. Can you optimize these?

Reply:
Thanks @vikasvb90 , I've made some optimization for it, the main point is to add a new field `dynamicFieldCount` to the class `MappingLookup`, when a new dynamic mapper is added, we increase the `dynamicFieldCount`, then we check if the current total fields count exceed the limit, so we don't need to calculate the dynamic field count from the dynamicMappers every time the check happens. Please help to take a look, thank you!